src/native/eventpipe/ep-provider.c

```diff
@@ -145,11 +145,13 @@ provider_compute_event_enable_mask (
 			if (session_provider) {
 				int64_t session_keyword = ep_session_provider_get_keywords (session_provider);
 				EventPipeEventLevel session_level = ep_session_provider_get_logging_level (session_provider);
+				// EventSources always set 0xF00000000000 to signify no keywords 
+				int64_t session_mask = ~0xF00000000000;
```

Comment:
Rather than do this filtering at the point of use, how about we filter these keyword bits at the point we are initializing the event in provider_add_event?